web/src/message_edit.js

```diff
@@ -1332,3 +1332,31 @@ export function with_first_message_id(stream_id, topic_name, success_cb, error_c
         error: error_cb,
     });
 }
+
+export function is_message_oldest_or_newest(
+    stream_id,
+    topic_name,
+    message_id,
+    success_callback,
+    error_callback,
+) {
+    const data = {
+        anchor: message_id,
+        num_before: 1,
+        num_after: 1,
+        narrow: JSON.stringify([
+            {operator: "stream", operand: stream_id},
+            {operator: "topic", operand: topic_name},
+        ]),
+    };
+
+    channel.get({
+        url: "/json/messages",
+        data,
+        success(data) {
+            const {found_oldest: is_oldest, found_newest: is_newest} = data;
+            success_callback(is_oldest, is_newest);
```

Comment:
Can you explain the logic here? I think `found_oldest` is true if the response contains the oldest message, but that doesn't speak to whether the message before or after is the oldest? I think we might need to actually parse the messages returned and just check if we got none before (first), none after (last), none (only), or 3 total messages (intermediate).

Reply:
During my testing, I found that when we query with the second last message as the anchor, found_newest is returned as false. Similarly, when the second oldest message is taken as the anchor, found_oldest is returned as false. Due to this behavior of the get messages API, my current logic works.

Reply:
I think this is still relying on what is intended to be an unspecified implementation detail of the server that could change; I'd prefer to inspect the `messages` part of the response. It should be pretty simple code: `is_oldest` is true if and only if the response doesn't contain any messages with a lower message ID, etc.

Reply:
Sure, I'll change the code to reflect the same. Thank you for pointing this out and I understand that we don't want to rely on a hacky solution that could change anytime.